src/deepsparse/pipeline.py

```diff
@@ -176,29 +177,33 @@ def __init__(
 
         self._batch_size = self._batch_size or 1
 
-    def __call__(self, *args, **kwargs) -> BaseModel:
+    def __call__(self, *args, monitoring: bool = False, **kwargs) -> BaseModel:
         if "engine_inputs" in kwargs:
             raise ValueError(
                 "invalid kwarg engine_inputs. engine inputs determined "
                 f"by {self.__class__.__qualname__}.parse_inputs"
             )
+        timer = TimingBuilder()
 
         # parse inputs into input_schema
+        timer.start(InferencePhases.TOTAL_INFERENCE.value)
```

Comment:
let's add support in timer.start/end to unwrap the enums so a user calling in won't have to. (ie `if isinstance(phase, Enum)...`

Reply:
Alternatively, I think we don't even need the `.value`?

```python
>>> from pydantic import BaseModel
>>> from enum import Enum
>>> class A(str, Enum):
...     a = "a"
...     b = "b"
... 
>>> class Schema(BaseModel):
...     a: int
...     b: int
... 
>>> {A.a: 1, A.b: 2}
{<A.a: 'a'>: 1, <A.b: 'b'>: 2}
>>> Schema(**{A.a: 1, A.b: 2})
Schema(a=1, b=2)
```